examples/rvc-app/rvc-common/src/rvc-device.cpp

```diff
@@ -6,11 +6,75 @@ using namespace chip::app::Clusters;
+                                                DataModel::Nullable<ServiceArea::PositionTag>(),
+                                                DataModel::Nullable<ServiceArea::FloorSurfaceTag>() );
+
+
+    // Location B has name, uses map XX
+    mServiceAreaInstance.AddSupportedLocation(  supportedLocationId_B, 
+                                                DataModel::Nullable<uint_fast8_t>(supportedMapId_XX),
+                                                CharSpan::fromCharString("My Location B"),
+                                                DataModel::Nullable<int16_t>(),
+                                                DataModel::Nullable<ServiceArea::AreaTypeTag>(),
+                                                DataModel::Nullable<ServiceArea::LandmarkTag>(),
+                                                DataModel::Nullable<ServiceArea::PositionTag>(),
+                                                DataModel::Nullable<ServiceArea::FloorSurfaceTag>() );
+
+
+
+
+   // Location C has full SemData, no name, Map YY
+    mServiceAreaInstance.AddSupportedLocation(  supportedLocationId_C, 
+                                                DataModel::Nullable<uint_fast8_t>(supportedMapId_YY),
+                                                CharSpan(),
+                                                DataModel::Nullable<int16_t>(-1),
+                                                DataModel::Nullable<ServiceArea::AreaTypeTag>(ServiceArea::AreaTypeTag::kPlayRoom),
+                                                DataModel::Nullable<ServiceArea::LandmarkTag>(ServiceArea::LandmarkTag::kBackDoor),
+                                                DataModel::Nullable<ServiceArea::PositionTag>(ServiceArea::PositionTag::kNextTo),
+                                                DataModel::Nullable<ServiceArea::FloorSurfaceTag>(ServiceArea::FloorSurfaceTag::kConcrete) );
+
+    // Location D has null values for all HomeLocationStruct fields, Map YY
+    mServiceAreaInstance.AddSupportedLocation(   supportedLocationId_D, 
+                                                DataModel::Nullable<uint_fast8_t>(supportedMapId_YY),
+                                                CharSpan::fromCharString("My Location D"),
+                                                DataModel::Nullable<int16_t>(),
+                                                DataModel::Nullable<ServiceArea::AreaTypeTag>(),
+                                                DataModel::Nullable<ServiceArea::LandmarkTag>(ServiceArea::LandmarkTag::kCouch),
+                                                DataModel::Nullable<ServiceArea::PositionTag>(ServiceArea::PositionTag::kNextTo),
+                                                DataModel::Nullable<ServiceArea::FloorSurfaceTag>(ServiceArea::FloorSurfaceTag::kHardwood) );
+
+
+    mServiceAreaInstance.SetCurrentLocation(supportedLocationId_C);
```

Comment:
I double [inky swear that I always compile and test before checking in!
But apparently this line was a last minute thought I slipped in.

In order to compile, needs to be

mServiceAreaInstance.SetCurrentLocation(DataModel::MakeNullable(supportedLocationId_C));

Reply:
Interesting. In the dyson repository this compiles without any changes.  Guess I did actually test it.
It fails to compile using the 1.3.0.0 version of connectedhomeip.

library/utility must have added implicit support for conversion to a Nullable Object.